Source/cmGlobalUnixMakefileGenerator3.cxx

```diff
@@ -179,15 +179,22 @@ void cmGlobalUnixMakefileGenerator3::Generate()
 
 void cmGlobalUnixMakefileGenerator3::AddLinkCommand(
   const std::vector<std::string>& sourceFiles,
-  const std::string& workingDirectory, const std::string& linkCommand)
+  const std::string& workingDirectory, const std::string& linkCommand,
+  const std::string& linkCommandsVersion)
 {
   if (!this->LinkCommandDatabase) {
+    Json::Value versionJsonValue;
+
     std::string commandDatabaseName =
       this->GetCMakeInstance()->GetHomeOutputDirectory() +
       "/link_commands.json";
+
     this->LinkCommandDatabase =
       cm::make_unique<cmGeneratedFileStream>(commandDatabaseName);
+    versionJsonValue["version"] = linkCommandsVersion;
```

Comment:
You already have this logic. Is there a way to not duplicate it?

Reply:
are you about cmGlobalNinjaGenerator and cmGlobalUnixMakefileGenerator3 class?